src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-form/rbd-form.component.spec.ts

```diff
@@ -78,10 +83,83 @@ describe('RbdFormComponent', () => {
   describe('test image configuration component', () => {
     it('is visible', () => {
       fixture.detectChanges();
-      expect(
-        fixture.debugElement.query(By.css('cd-rbd-configuration-form')).nativeElement.parentElement
-          .hidden
-      ).toBe(false);
+      expect(queryNativeElement('cd-rbd-configuration-form').parentElement.hidden).toBe(false);
     });
   });
+
+  describe('tests for feature flags', () => {
+    let deepFlatten, layering, exclusiveLock, objectMap, journaling, fastDiff;
+
+    const setFeatures = (features) => {
+      component.features = features;
+      component.featuresList = component.objToArray(features);
+      component.createForm();
+    };
+
+    const resetFeatures = (
+      defaultFeatures = ['deep-flatten', 'exclusive-lock', 'fast-diff', 'layering', 'object-map']
+    ) => {
+      const rbdService = TestBed.get(RbdService);
+      spyOn(rbdService, 'defaultFeatures').and.returnValue(of(defaultFeatures));
+      component.router = { url: '/block/rbd/create' } as Router;
+      component.ngOnInit();
+      tick();
+      fixture.detectChanges();
```

Comment:
`fixture.detectChanges` will call `component.ngOnInit` if called, so it's not necessary to call `ngOnInit` before.

Reply:
Thanks, this is good to know!